packages/contracts-bedrock/test/safe/SaferSafes.t.sol

```diff
@@ -0,0 +1,144 @@
+            fallbackOwner: fallbackOwner
+        });
+
+        vm.prank(address(safeInstance.safe));
+        saferSafes.configureLivenessModule(moduleConfig);
+
+        // Configure the timelock guard SECOND (this will trigger the check)
+        vm.prank(address(safeInstance.safe));
+        saferSafes.configureTimelockGuard(timelockDelay);
+
+        // Verify configurations were set
+        (uint256 storedLivenessResponsePeriod, address storedFallbackOwner) =
+            saferSafes.livenessSafeConfiguration(address(safeInstance.safe));
+        assertEq(storedLivenessResponsePeriod, livenessResponsePeriod);
+        assertEq(storedFallbackOwner, fallbackOwner);
+    }
+
+    function test_configure_timelockGuardFirst_succeeds() public {
+        uint256 timelockDelay = 7 days;
+        uint256 livenessResponsePeriod = 21 days; // Much greater than 2 * 7 days = 14 days (should succeed)
+
+        // Configure the timelock guard FIRST
+        vm.prank(address(safeInstance.safe));
+        saferSafes.configureTimelockGuard(timelockDelay);
+
+        LivenessModule2.ModuleConfig memory moduleConfig = LivenessModule2.ModuleConfig({
+            livenessResponsePeriod: livenessResponsePeriod,
+            fallbackOwner: fallbackOwner
+        });
+
+        // Configure the liveness module SECOND (this will trigger the check)
+        vm.prank(address(safeInstance.safe));
+        saferSafes.configureLivenessModule(moduleConfig);
+
+        // Verify configurations were set
+        (uint256 storedLivenessResponsePeriod, address storedFallbackOwner) =
+            saferSafes.livenessSafeConfiguration(address(safeInstance.safe));
+        assertEq(storedLivenessResponsePeriod, livenessResponsePeriod);
+        assertEq(storedFallbackOwner, fallbackOwner);
+    }
+
+    /// @notice Test that attempting to enable the second component with incompatible config fails.
+    /// @dev This test would fail if timelock guard configuration also triggered validation
+    function test_configure_livenessModuleFirstInvalidConfig_reverts() public {
+        uint256 timelockDelay = 7 days;
+        uint256 livenessResponsePeriod = 13 days; // This is invalid: 13 < 2*7
+
+        // Configure liveness module first
+        LivenessModule2.ModuleConfig memory moduleConfig = LivenessModule2.ModuleConfig({
+            livenessResponsePeriod: livenessResponsePeriod,
+            fallbackOwner: fallbackOwner
+        });
+
+        vm.prank(address(safeInstance.safe));
+        saferSafes.configureLivenessModule(moduleConfig);
+
+        // Now configure timelock guard - this depends on when validation is triggered
+        // Based on current behavior, this should succeed because validation doesn't happen during reconfiguration
+        vm.prank(address(safeInstance.safe));
+        vm.expectRevert(SaferSafes.SaferSafes_InsufficientLivenessResponsePeriod.selector);
+        saferSafes.configureTimelockGuard(timelockDelay);
```

Comment:
The comment seems to contradict the test here. The comment says this should succeed, but then we expect a revert.